docs/dev/contributing.rst

```diff
@@ -0,0 +1,78 @@
+Workflow
+========
+
+Recommended one-time setup
+--------------------------
+
+Remove ``--global`` if you don't want it to apply to other git repos on your computer: ::
+
+  git config --global user.name "Your Name"
+  git config --global user.email you@something.edu
+
+Other git config you might find useful: ::
+
+  git config --global push.default simple # makes git only push the current branch, which is useful for not accidentally messing things up
+  git config --global color.{ui,status,branch,diff,interactive} auto # makes various interfaces usefully colorful
+  git config --global log.decorate true # shows branch information in `git log` by default
+
+For normal, non-urgent features and bug fixes
+---------------------------------------------
+
+The following workflow applies if you've already been added as a collaborator to the repository.  If not, you should fork it using the button on Github, add it as a remote, and replace all of the ``git push`` steps with ``git push <remote name>``.
+
+From the directory ``/esp``: ::
+
+  git checkout main
+  git pull
+  ./update_deps.sh # on vagrant: see vagrant docs; no need to bother if deps haven’t changed
+  ./manage.py update # on vagrant: fab manage:cmd=update
+  git checkout -b new-branch-name
```

Comment:
Some places in this doc put custom branch/remote names in angle brackets, and some don't. We should be consistent.